src/main/java/graphql/execution/ExecutionStrategy.java

```diff
@@ -680,20 +680,11 @@ private Object handleCoercionProblem(ExecutionContext context, ExecutionStrategy
     }
 
 
-    /**
-     * Converts an object that is known to should be an Iterable into one
-     *
-     * @param result the result object
-     *
-     * @return an Iterable from that object
-     *
-     * @throws java.lang.ClassCastException if it's not an Iterable
-     */
-    protected Iterable<Object> toIterable(Object result) {
-        return FpKit.toIterable(result);
-    }
-
     protected GraphQLObjectType resolveType(ExecutionContext executionContext, ExecutionStrategyParameters parameters, GraphQLType fieldType) {
+        // we can avoid a method call and type resolver environment allocation if we know it's an object type
+        if (fieldType instanceof GraphQLObjectType) {
+            return (GraphQLObjectType) fieldType;
+        }
```

Comment:
we can avoid calling the extra method and inside that the resolver type env allocation if we know its an object type upfront

I originally did this inside `esolvedType.resolveType()` but why even call it  with object types.  This is faster